### ISSUE TITLE
New DataSource and Resource: aws_db_cluster_snapshot

### ISSUE THREAD

**Issue by 4dz** [terraform-providers/terraform-provider-aws#4525]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

There is a useful aws_db_snapshot data source for querying RDS snapshots.
But I cannot see an equivalent for DB Cluster Snapshots (such as Aurora).

Is there a reason it cannot be done or has no one requested it (until now)?

These are useful for the case where you have a dev environment which you want to destroy and later recreate, e.g. for projects which are not under constant development.

### New or Affected Resource(s)

* aws_db_cluster_snapshot

### Potential Terraform Configuration

```hcl
data "aws_db_cluster_snapshot" "latest_dev_final_snapshot" {
  db_cluster_identifier = "development"
  most_recent = true
  snapshot_type = "automatic|manual"
}

data "aws_db_cluster_snapshot" "latest_prod_snapshot" {
  db_cluster_snapshot_identifier = "${ ... }"
}

resource "aws_db_cluster_snapshot" "test" {
	db_cluster_identifier = "${aws_rds_cluster.aurora.id}"
	db_cluster_snapshot_identifier = "testsnapshot%d"
}
```

### References

* https://www.terraform.io/docs/providers/aws/d/db_snapshot.html
* https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBClusterSnapshots.html

* Probably relevant, https://github.com/hashicorp/terraform/pull/10291

**Comment by 4dz** [comment]
I forgot to mention the corresponding resource to complement the data source!

**Comment by 4dz** [comment]
I'd like to try to add a pull request but I've not written Go before, will give it a go as its so similar to the existing aws_db_snapshot

**Comment by bflad** [comment]
Thanks to @4dz the new `aws_db_cluster_snapshot` resource and data source will be available in version 1.31.0 of the AWS provider, likely releasing later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.31.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1310-august-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
